app/front-end/src/features/editor/utils/helpers.tsx

```diff
@@ -90,3 +90,20 @@ export const getWorkspaceArray = (fileTreeView: TreeViewBaseItem<FileTreeViewIte
 
   return workspaceArray;
 };
+
+export const generateTimestamp = () => {
+  const now = new Date();
+  return (
+    now.getFullYear() +
+    '-' +
+    String(now.getMonth() + 1).padStart(2, '0') + // Months are zero-indexed
+    '-' +
+    String(now.getDate()).padStart(2, '0') +
+    '_' +
+    String(now.getHours()).padStart(2, '0') +
+    ':' +
+    String(now.getMinutes()).padStart(2, '0') +
+    ':' +
```

Comment:
':' is an illegal file name character in Windows. Would be better and safer to just seperate everything with '-' or '_'.